play23/src/main/scala/play/api/test/ops/AsyncResultExtractors.scala

```diff
@@ -0,0 +1,123 @@
+package play.api.test.ops
+
+import play.api.http.HeaderNames._
+import play.api.http.Status._
+import play.api.libs.iteratee.Iteratee
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc._
+import play.twirl.api.Content
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Similar to play.api.test.ResultExtractors except that it is designed to work better
+  * for org.scalatest.AsyncSuite in ScalaTest 3.0.
+  *
+  * All methods return either the expected value or a [[Future]] and does not perform
+  * any awaits or timeouts. This improves the total time spent on tests and cuts
+  * down on processor usage.
+  */
+trait AsyncResultExtractors {
+
+  /**
+    * Extracts the Content-Type of this Content value.
+    */
+  def contentType(of: Content): String = of.contentType
+
+  /**
+    * Extracts the content as String.
+    */
+  def contentAsString(of: Content): String = of.body
+
+  /**
+    * Extracts the content as bytes.
+    */
+  def contentAsBytes(of: Content): Array[Byte] = of.body.getBytes
+
+  /**
+    * Extracts the content as Json.
+    */
+  def contentAsJson(of: Content): JsValue = Json.parse(of.body)
+
+  /**
+    * Extracts the Content-Type of this Result value.
+    */
+  def contentType(result: Result): Option[String] = {
+    header(CONTENT_TYPE, result).map(_.split(";").take(1).mkString.trim)
+  }
+
+  /**
+    * Extracts the Charset of this Result value.
+    */
+  def charset(result: Result): Option[String] = header(CONTENT_TYPE, result) match {
+    case Some(s) if s.contains("charset=") => Some(s.split("; charset=").drop(1).mkString.trim)
```

Comment:
I copped this from Play... not sure why they do it that way:
https://github.com/playframework/playframework/blob/b692d1714b74864d9c86ab0708f3d349a5759072/framework/src/play-test/src/main/scala/play/api/test/Helpers.scala#L317-L320